### ISSUE TITLE
CA1862 code fix changes semantics for s.ToLower(...) != s (rewrites to self-compare that’s always false)

### ISSUE THREAD

**Issue by s2quake** [dotnet/roslyn-analyzers#7758]
## Issue Description ##

The analyzer suggests: “Prefer using 'string.Equals(string, StringComparison)' to perform a case-insensitive comparison … consider using 'StringComparison.OrdinalIgnoreCase'.”

For the pattern s.ToLower(CultureInfo.InvariantCulture) != s, the automatic fix produces !s.Equals(s, StringComparison.CurrentCultureIgnoreCase), which changes behavior.

* Original intent: detect if s contains any uppercase letters.
* After fix: compares s with itself ignoring case (always true) and negates it, resulting in false for all inputs.

This turns a meaningful condition into a no-op.
<img width="693" height="245" alt="Image" src="https://github.com/user-attachments/assets/8a76a95e-504a-4b9a-b812-d348611816d8" />>

## Steps to Reproduce ##

* Original code:

```csharp
using System.Globalization;

var hex = "0123456789ABcdefABcdEfABcdEFabcDEFabCDEF";
if (hex.ToLower(CultureInfo.InvariantCulture) != hex)
{
    Console.WriteLine("The string contains uppercase letters.");
}
```

* Apply CA1862 fix → becomes:

```csharp
if (!hex.Equals(hex, StringComparison.CurrentCultureIgnoreCase))
{
    Console.WriteLine("The string contains uppercase letters.");
}
```

## Expected Behavior ##

No fix should be offered when both sides of the comparison refer to the same variable/instance (self-comparison).
Alternatively, offer a semantics-preserving suggestion or guidance (e.g., recommend checking characters with hex.Any(char.IsUpper) if the goal is to detect uppercase).

## Actual Behavior ##

The offered fix rewrites to a self-equality check with ignore-case, which is always true, and then negates it, making the condition always false.

Rule message is appropriate for comparing two different strings case-insensitively, but the code fix should avoid triggering on self-comparisons like s.ToLower(...) != s.

### C# log ###
<details>Post the output from Output-->C# here</details>

### C# LSP Trace Logs ###
<details>Post the output from Output-->C# LSP Trace Logs here.  Requires the `C#` output window log level to be set to `Trace`</details>

## Environment information ##

**VSCode version**: 1.103.2
**C# Extension**: 2.87.31
**Using OmniSharp**: false

<details><summary>Dotnet Information</summary>
.NET SDK:
 Version:           9.0.203
 Commit:            dc7acfa194
 Workload version:  9.0.200-manifests.9df47798
 MSBuild version:   17.13.20+a4ef1e90f

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  15.6
 OS Platform: Darwin
 RID:         osx-arm64
 Base Path:   /usr/local/share/dotnet/sdk/9.0.203/

.NET workloads installed:
There are no installed workloads to display.
Configured to use loose manifests when installing new manifests.

Host:
  Version:      9.0.4
  Architecture: arm64
  Commit:       f57e6dc747

.NET SDKs installed:
  8.0.408 [/usr/local/share/dotnet/sdk]
  9.0.203 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 8.0.15 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 9.0.4 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 8.0.15 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 9.0.4 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  x64   [/usr/local/share/dotnet]
    registered at [/etc/dotnet/install_location_x64]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|Folder Name|
|---|---|---|---|
|copilot|GitHub|1.364.0|github.copilot-1.364.0|
|copilot-chat|GitHub|0.30.3|github.copilot-chat-0.30.3|
|csdevkit|ms-dotnettools|1.50.33|ms-dotnettools.csdevkit-1.50.33-darwin-arm64|
|csharp|ms-dotnettools|2.87.31|ms-dotnettools.csharp-2.87.31-darwin-arm64|
|EditorConfig|EditorConfig|0.17.4|editorconfig.editorconfig-0.17.4|
|gitlens|eamodio|17.4.1|eamodio.gitlens-17.4.1|
|go|golang|0.50.0|golang.go-0.50.0|
|hexeditor|ms-vscode|1.11.1|ms-vscode.hexeditor-1.11.1|
|ilspy-vscode|icsharpcode|0.22.0|icsharpcode.ilspy-vscode-0.22.0|
|java|redhat|1.45.0|redhat.java-1.45.0-darwin-arm64|
|markdown-all-in-one|yzhang|3.6.3|yzhang.markdown-all-in-one-3.6.3|
|powershell|ms-vscode|2025.2.0|ms-vscode.powershell-2025.2.0|
|remote-containers|ms-vscode-remote|0.422.1|ms-vscode-remote.remote-containers-0.422.1|
|rome|rome|0.28.0|rome.rome-0.28.0-darwin-x64|
|vscode-avalonia|AvaloniaTeam|0.0.32|avaloniateam.vscode-avalonia-0.0.32|
|vscode-dotnet-runtime|ms-dotnettools|2.3.7|ms-dotnettools.vscode-dotnet-runtime-2.3.7|
|vscode-github-actions|github|0.27.2|github.vscode-github-actions-0.27.2|
|vscode-graphql|GraphQL|0.13.2|graphql.vscode-graphql-0.13.2|
|vscode-graphql-syntax|GraphQL|1.3.8|graphql.vscode-graphql-syntax-1.3.8|
|vscode-nupkg|eridem|1.0.1|eridem.vscode-nupkg-1.0.1|
|vscode-proto3|zxh404|0.5.5|zxh404.vscode-proto3-0.5.5|
|vscode-pull-request-github|GitHub|0.116.2|github.vscode-pull-request-github-0.116.2|
|vscode-yaml|redhat|1.18.0|redhat.vscode-yaml-1.18.0|
|vscode-zipfs|arcanis|3.0.0|arcanis.vscode-zipfs-3.0.0|
|vstuc|visualstudiotoolsforunity|1.1.3|visualstudiotoolsforunity.vstuc-1.1.3|;

</details>
